src/phoenix/server/api/helpers/playground_spans.py

```diff
@@ -224,12 +224,49 @@ def _input_value_and_mime_type(input: Any) -> Iterator[tuple[str, Any]]:
     yield INPUT_VALUE, safe_json_dumps(input_data)
 
 
-def _output_value_and_mime_type(output: Any) -> Iterator[tuple[str, Any]]:
-    yield OUTPUT_MIME_TYPE, JSON
-    yield OUTPUT_VALUE, safe_json_dumps(jsonify(output))
+def _merge_tool_call_chunks(
+    chunks_by_id: defaultdict[str, list[ToolCallChunk]],
+) -> Union[list[dict[str, Any]], dict[str, Any]]:
+    merged_tool_calls = []
+
+    for tool_id, chunks in chunks_by_id.items():
+        # Get the name from the first chunk since it should be the same for all chunks
+        first_chunk = chunks[0]
+
+        # Combine all argument chunks
+        merged_arguments = "".join(chunk.function.arguments for chunk in chunks)
+
+        merged_tool_calls.append(
+            {
+                "id": tool_id,
+                "function": {"name": first_chunk.function.name, "arguments": merged_arguments},
+            }
+        )
+
+    return merged_tool_calls[0] if len(merged_tool_calls) == 1 else merged_tool_calls
+
+
+def _output_value_and_mime_type(
+    text_chunks: list[TextChunk],
+    tool_call_chunks: defaultdict[ToolCallID, list[ToolCallChunk]],
+) -> Iterator[tuple[str, Any]]:
+    content = "".join(chunk.content for chunk in text_chunks)
+    merged_tool_calls = _merge_tool_call_chunks(tool_call_chunks)
+    if len(merged_tool_calls) == 0:
+        yield OUTPUT_MIME_TYPE, TEXT
+        yield OUTPUT_VALUE, content
+    elif len(content) == 0:
+        yield OUTPUT_MIME_TYPE, JSON
+        yield OUTPUT_VALUE, safe_json_dumps(jsonify(merged_tool_calls))
+    else:
+        yield OUTPUT_MIME_TYPE, JSON
+        yield (
+            OUTPUT_VALUE,
+            safe_json_dumps({"content": content, "tool_calls": jsonify(merged_tool_calls)}),
+        )
```

Comment:
Non-blocking suggestion: This might read a little clearer if we create an output dictionary, add to it, and return it.